### ISSUE TITLE
Missing / excessive diagnostics for preprocessing directives

### ISSUE THREAD

**Issue by pmor13** [cc65/cc65#2515]
Compiling this code:
```
#line 13"x"
```
leads to:
```
Error: #line directive requires a simple decimal digit sequence
```
while it does not violate any syntax rule or constraint.

Compiling this code:
```
#define X"y"
```
leads to:
```
<nothing>
```
while it violates constraint "There shall be white-space between the identifier and the replacement list in the definition
of an object-like macro." A conforming implementation shall produce at least one diagnostic message.

**Comment by mrdudz** [comment]
> while it violates constraint "There shall be white-space between the identifier and the replacement list in the definition
of an object-like macro."

To be exact: in C99 mode

**Comment by mrdudz** [comment]
fixed via #2518
